# 03 - Estrutura de Repetição

**Exercício 01**

Faça um programa que peça uma nota, entre zero e dez. Mostre uma mensagem caso o valor seja inválido e continue pedindo até que o usuário informe um valor válido.

In [6]:
# while True:
#     nota = float(input("Informe uma nota entre 0 e 10: "))
#     if nota <= 0 or nota > 10:
#         continue
#     else:
#         print(f'Nota informada: {nota}')
#         break

# Solução 2
while True:
    try:
        numero = int(input('Digite um valor de 0 a 10: '))
    except ValueError:
        print('Deve ser fornecido um valor inteiro')
    else:
        if 0 <= numero <= 10: 
            print(f'Número informado é: {numero}')
            break
        else:
            print('O número deve estar entre 0 e 10.')



Deve ser fornecido um valor inteiro
Deve ser fornecido um valor inteiro
O número deve estar entre 0 e 10.
Número informado é: 6


**Exercício 02**

Faça um programa que leia um nome de usuário e a sua senha e não aceite a senha igual ao nome do usuário, mostrando uma mensagem de erro e voltando a pedir as informações.

In [1]:
while True:
    usuario = input('Nome de usuário: ')
    senha = input('Senha: ')
    if usuario == senha:
        print("Usuário não pode ser igual a senha")
    else:
        print('Bem-vindo!')
        break


Usuário não pode ser igual a senha


**Exercício 03**

Faça um programa que leia e valide as seguintes informações:

a) Nome: maior que 3 caracteres;  
b) Idade: entre 0 e 150;  
c) Salário: maior que zero;  
d) Sexo: 'f' ou 'm';  
e) Estado Civil: 's', 'c', 'v', 'd';

In [7]:
while True:
    nome = input("Informe o nome: ")
    if len(nome) <= 3:
        print('Nome deve ter mais que 3 caracteres.')
    else:
        print(f'Nome: {nome}')
        break

while True:
    try:
        idade = int(input("Informe a idade: "))
    except ValueError:
        print('Deve ser fornecido um valor inteiro')
    else:
        if 0 <= idade <= 150: 
            print(f'Idade: {idade}')
            break
        else:
            print('A idade deve estar entre 0 e 150.')

while True:
    try:
        salario = float(input('Informe o salário: '))
    except ValueError:
        print('Deve ser fornecido um valor real.')
    else:
        if salario > 0:
            print(f'Salário: {salario}')
            break
        else:
            print('O salário deve ser maior que 0.')
        
while True:
    sexo = input('Informe o sexo (M ou F): ').upper()
    if sexo == 'M' or 'F':
        print(f'Sexo: {sexo}')
        break
    else:
        print('Informe o sexo como M ou F.')
    
while True:
    estado_civil = input('Informe o estado civil (S, C, V, D): ').upper()
    if estado_civil not in ['S', 'C', 'V', 'D']:
        print('Estado civil deve ser S, C, V ou D')
    else:
        print(f'Estado Civil: {estado_civil}')
        break


Nome deve ter mais que 3 caracteres.
Nome: joao
Idade: 35
Salário: 10000.0
Sexo: M
Estado Civil: V


**Exercício 04**

Supondo que a população de um país A seja da ordem de 80.000 habitantes com uma taxa anual de crescimento de 3% e que a população de B seja 200.000 habitantes com uma taxa de crescimento de 1.5%. Faça um programa que calcule e escreva o número de anos necessários para que a população do país A ultrapasse ou iguale a população do país B, mantidas as taxas de crescimento.

In [2]:
populacao_a = 80_000
populacao_b = 200_000
taxa_a = 3 / 100
taxa_b = 1.5 / 100
anos = 0

while populacao_b > populacao_a:
    anos += 1
    populacao_a = int(populacao_a + populacao_a * taxa_a)
    populacao_b = int(populacao_b + populacao_b * taxa_b)

print(f'''
População A: {populacao_a}
População B: {populacao_b}
Anos: {anos}
''')



População A: 514954
População B: 510914
Anos: 63



**Exercício 05**

Altere o programa anterior permitindo ao usuário informar as populações e as taxas de crescimento iniciais. Valide a entrada e permita repetir a operação.

In [8]:
while True:    
    while True:
        try:
            populacao_a = int(input("Informe a população A: "))
            break
        except ValueError:
            print('Informe um valor inteiro.')

    while True:
        try:
            taxa_de_crescimento_de_a = float(input("Informe a taxa de crescimento de A (2 para 2%): ")) / 100 + 1
            break
        except ValueError:
            print('Informe um valor numérico real.')

    while True:
        try:
            populacao_b = int(input("Informe a população B: "))
            break
        except ValueError:
            print('Informe um valor inteiro.')

    while True:
        try:
            taxa_de_crescimento_de_b = float(input("Informe a taxa de crescimento de B (2 para 2%): ")) / 100 + 1
            break
        except ValueError:
            print('Informe um valor numérico real.')

    anos = 0

    while populacao_a <= populacao_b:
        anos += 1
        populacao_a *= taxa_de_crescimento_de_a
        populacao_b *= taxa_de_crescimento_de_b

    print(f'''
    População A: {populacao_a}
    População B: {populacao_b}
    Anos: {anos}
    ''')

    print('Digite 0 sair ou qualquer outra tecla para continuar')
    escolha = input()
    if escolha == "0":
        break


    População A: 515033.1028059676
    População B: 510964.4169622198
    Anos: 63
    
Digite 0 sair ou qualquer outra tecla para continuar
Informe um valor inteiro.
Informe um valor inteiro.

    População A: 1275774.2886835756
    População B: 1272932.9231707724
    Anos: 186
    
Digite 0 sair ou qualquer outra tecla para continuar


KeyboardInterrupt: 

**Exercício 06**

Faça um programa que imprima na tela os números de 1 a 20, um abaixo do outro. Depois modifique o programa para que ele mostre os números um ao lado do outro.

In [10]:
for numero in range(1, 21):
    print(numero)

for numero in range(1, 21):
    print(numero, end=' ')

1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 

**Exercício 07**

Faça um programa que leia 5 números e informe o maior número.

In [12]:
# numero_1 = 20
# numero_2 = 30
# numero_3 = 10
# numero_4 = 0
# numero_5 = 2

# numeros = [numero_1, numero_2, numero_3, numero_4, numero_5]
# maior = numeros[0]

# for numero in numeros:
#     if numero > maior:
#         maior = numero

# print(f'Maior: {maior}')


# Solução 2

# _ significa que não estamos usando aquela variável
maximo = float(input('Digite um número: '))
for _ in range(4):
    maximo = max(maximo, float(input('Digite um número: ')))
    print(f'Número máximo encontrado até agora: {maximo}')



Maior: 30


**Exercício 08**

Faça um programa que leia 5 números e informe a soma e a média dos números.

In [15]:
# solução 1
# soma = 0

# for _ in range(5):
#     soma += float(input("Digite um número: "))

# media = soma / 5
# print(f'Soma: {soma}')
# print(f'Média: {media}')


# Solução 2

soma = float(input('Digite um número 1: '))
for n in range(2, 6):
    soma += float(input(f'Digite o número {n}: '))
    media = soma / n
    print(f'A soma dos números é: {soma}, e a média é {media}')


A soma dos números é: 12.0, e a média é 6.0
A soma dos números é: 101.0, e a média é 33.666666666666664
A soma dos números é: 202.0, e a média é 50.5
A soma dos números é: 402.0, e a média é 80.4


**Exercício 09**

Faça um programa que imprima na tela apenas os números ímpares entre 1 e 50.

In [16]:
for numero in range(1, 51, 2):
    print(numero, end=' ')

1 3 5 7 9 11 13 15 17 19 21 23 25 27 29 31 33 35 37 39 41 43 45 47 49 

**Exercício 10**

Faça um programa que receba dois números inteiros e gere os números inteiros que estão no intervalo compreendido por eles.

In [17]:
limite_inferior = int(input("Informe o limite inferior: "))
limite_superior = int(input("Informe o limite superior: "))
for numero in range(limite_inferior, limite_superior + 1):
    print(numero, end = ' ')

5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 

**Exercício 11**

Altere o programa anterior para mostrar no final a soma dos números.

In [20]:
limite_inferior = int(input("Informe o limite inferior: "))
limite_superior = int(input("Informe o limite superior: "))
soma = 0
for numero in range(limite_inferior, limite_superior + 1):
    soma += numero
    print(numero, end = ' ')

print(f'\nSoma: {soma}')

1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95 96 97 98 99 100 
Soma: 5050


**Exercício 12**

Desenvolva um gerador de tabuada, capaz de gerar a tabuada de qualquer número inteiro entre 1 a 10. O usuário deve informar de qual numero ele deseja ver a tabuada. A saída deve ser conforme o exemplo abaixo:

Tabuada de 5:  
5 X 1 = 5  
5 X 2 = 10  
...  
5 X 10 = 50

In [4]:
tabuada = int(input('Informe de que número deseja ver a tabuada: '))

for i in range(1, 11):
    print(f'{tabuada} x {i} = {tabuada * i}')

9 x 1 = 9
9 x 2 = 18
9 x 3 = 27
9 x 4 = 36
9 x 5 = 45
9 x 6 = 54
9 x 7 = 63
9 x 8 = 72
9 x 9 = 81
9 x 10 = 90


**Exercício 13**

Faça um programa que peça dois números, base e expoente, calcule e mostre o primeiro número elevado ao segundo número. Não utilize a função de potência da linguagem.

In [10]:
base = int(input('Informe o valor da base: '))
expoente = int(input('Informe o valor do expoente: '))

resultado = 1

for _ in range(expoente):
    resultado *= base

print(f'{base} elevado a {expoente} é igual a {resultado}')


2 elevado a 10 é igual a 1024


**Exercício 14**

Faça um programa que peça 10 números inteiros, calcule e mostre a quantidade de números pares e a quantidade de números impares.

In [1]:
pares = 0
impares = 0
for i in range(10):
    numero = int(input(f'Informe o número {i + 1}: '))
    if numero % 2 == 0:
        pares += 1
    else:
        impares += 1

print(f'''
Pares: {pares}
Ímpares: {impares}
''')


Pares: 4
Ímpares: 6



**Exercício 15**

A série de Fibonacci é formada pela seqüência 1, 1, 2, 3, 5, 8, 13, 21, 34, 55,... Faça um programa capaz de gerar a série até o n−ésimo termo.

**Exercício 16**

A série de Fibonacci é formada pela seqüência 0,1,1,2,3,5,8,13,21,34,55,... Faça um programa que gere a série até que o valor seja maior que 500.

**Exercício 17**

Faça um programa que calcule o fatorial de um número inteiro fornecido pelo usuário. Ex.: 5!=5.4.3.2.1=120

In [2]:
n = 5

fatorial = 1

for i in range(1, n + 1):
    fatorial *= i

print(f'{n}! = {fatorial}')

5! = 120


**Exercício 18**

Faça um programa que, dado um conjunto de N números, determine o menor valor, o maior valor e a soma dos valores.

In [4]:
n = 5
numeros = []
soma = 0

for i in range (1, n + 1):
    numero = int(input(f'Informe o {i}º número: '))
    numeros.append(numero)

menor = numeros[0]
maior = numeros[0]

for numero in numeros:
    menor = min(numero, menor)
    maior = max(numero, maior)
    soma += numero

print(f'''
Números: {numeros}
Menor: {menor}
Maior: {maior}
Soma: {soma}
''')




Números: [1, 2, 3, 4, 5]
Menor: 1
Maior: 5
Soma: 15



**Exercício 19**

Altere o programa anterior para que ele aceite apenas números entre 0 e 1000.

In [7]:
n = 5
numeros = []
soma = 0
contador = 1

while contador <= n:
    numero = int(input(f'Informe o {contador}º número: '))
    if 0 <= numero <= 1000:
        numeros.append(numero)
        contador += 1
    else:
        print('Informe apenas números entre 0 e 1000')
        continue

menor = numeros[0]
maior = numeros[0]

for numero in numeros:
    menor = min(numero, menor)
    maior = max(numero, maior)
    soma += numero

print(f'''
Números: {numeros}
Menor: {menor}
Maior: {maior}
Soma: {soma}
''')


Informe apenas números entre 0 e 1000
Informe apenas números entre 0 e 1000

Números: [1, 5, 3, 9, 7]
Menor: 1
Maior: 9
Soma: 25



**Exercício 20**

Altere o programa de cálculo do fatorial, permitindo ao usuário calcular o fatorial várias vezes e limitando o fatorial a números inteiros positivos e menores que 16.

In [1]:
while True:
    n = int(input('Digite o número que deseja calcular o fatorial (0 para sair): '))
    if n == 0:
        break
    elif 0 < n <= 16:

        fatorial = 1

        for i in range(1, n + 1):
            fatorial *= i

        print(f'{n}! = {fatorial}')
    else:
        print('Apenas números inteiros positivos menores que 16.')
        continue

9! = 362880
1! = 1
5! = 120
Apenas números inteiros positivos menores que 16.
3! = 6


**Exercício 21**

Faça um programa que peça um número inteiro e determine se ele é ou não um número primo. Um número primo é aquele que é divisível somente por ele mesmo e por 1.

In [7]:
numero = int(input('Digite um número inteiro: '))

primo = True

for i in range (2, numero):
    if numero % i == 0:
        primo = False

if primo:
    print(f'{numero} é primo')
else:
    print(f'{numero} não é primo')


29 é primo


**Exercício 22**

Altere o programa de cálculo dos números primos, informando, caso o número não seja primo, por quais número ele é divisível.

In [11]:
numero = int(input('Digite um número inteiro: '))

divisores = []
primo = True

for i in range (2, numero):
    if numero % i == 0:
        divisores.append(i)
        primo = False

if primo:
    print(f'{numero} é primo')
else:
    print(f'{numero} não é primo')
    print('Divisores:')
    for divisor in divisores:
        print(divisor, end=' ')

230 não é primo
Divisores:
2 5 10 23 46 115 

**Exercício 23**

Faça um programa que mostre todos os primos entre 1 e N sendo N um número inteiro fornecido pelo usuário. O programa deverá mostrar também o número de divisões que ele executou para encontrar os números primos. Serão avaliados o funcionamento, o estilo e o número de testes (divisões) executados.

**Exercício 24**

Faça um programa que calcule o mostre a média aritmética de N notas.

In [15]:
contador = 1
soma = 0
media = 0

while True:
    nota = float(input(f'Informe a nota ou -1 para parar. {contador}ª nota: '))
    if nota == -1:
        break
    soma += nota
    media = soma / contador
    contador += 1

print(f'Notas: {contador - 1}')
print(f'Notas: {media}')


Notas: 3
Notas: 2.5


**Exercício 25**

Faça um programa que peça para n pessoas a sua idade, ao final o programa devera verificar se a média de idade da turma varia entre 0 e 25, 26 e 60 e maior que 60; e então, dizer se a turma é jovem, adulta ou idosa, conforme a média calculada.

In [19]:
pessoas = int(input('Informe quantas pessoas: '))
soma = 0

for i in range(1, pessoas + 1):
    idade = int(input(f'Idade da pessoa {i}: '))
    soma += idade

media = soma / pessoas

if 0 <= media <= 25:
    turma = 'jovem'
elif 26 <= media <= 60:
    turma = 'adulta'
elif media > 60:
    turma = 'idosa'
else:
    print('valor inválido.')

print(f'''
Pessoas: {pessoas}
Média de idade: {media:.2f}
Conclusão: turma {turma}
''')


Pessoas: 2
Média de idade: 8.50
Conclusão: turma jovem



**Exercício 26**

Numa eleição existem três candidatos. Faça um programa que peça o número total de eleitores. Peça para cada eleitor votar e ao final mostrar o número de votos de cada candidato.

In [21]:
eleitores = 10

a = 0
b = 0
c = 0

for i in range(1, eleitores + 1):
    while True:
        voto = input(f'{i}º eleitor, informe seu voto: {a, b, c}: ').lower()
        if voto not in ['a', 'b', 'c']:
            print('Voto inválido.')
            continue
        if voto == 'a':
            a += 1
            break
        elif voto == 'b':
            b += 1
            break
        elif voto == 'c':
            c += 1
            break

print(f'''
A: {a} votos
B: {b} votos
C: {c} votos
''')

Voto inválido.
Voto inválido.

A: 3 votos
B: 3 votos
C: 4 votos



**Exercício 27**

Faça um programa que calcule o número médio de alunos por turma. Para isto, peça a quantidade de turmas e a quantidade de alunos para cada turma. As turmas não podem ter mais de 40 alunos.

In [23]:
turmas = 3
soma = 0
for i in range(1, turmas + 1):
    while True:
        alunos = int(input(f'Informe o número de alunos da turma {i}: '))
        if alunos > 40:
            print('Turmas devem ter no máximo 40 alunos.')
            continue
        else:
            soma += alunos
            break

media = soma / turmas

print(f'Turmas: {turmas}')
print(f'Média de alunos por turma: {int(media)}')

Turmas devem ter no máximo 40 alunos.
Turmas devem ter no máximo 40 alunos.
Turmas: 3
Média de alunos por turma: 31


**Exercício 28**

Faça um programa que calcule o valor total investido por um colecionador em sua coleção de CDs e o valor médio gasto em cada um deles. O usuário deverá informar a quantidade de CDs e o valor para em cada um.

In [1]:
cds = int(input('Informe a quantidade de CDs: '))
soma = 0
for i in range(1, cds + 1):
    preco = float(input(f'Informe o preço do CD {i}: '))
    soma += preco

media = soma / cds

print(f'''
Quantidade de CDs: {cds}
Valor Investido: R$ {soma:.2f}
Valor Médio por CD: R$ {media:.2f}
''')




Quantidade de CDs: 5
Valor Investido: R$ 156.00
Valor Médio por CD: R$ 31.20



**Exercício 29**

O Sr. Manoel Joaquim possui uma grande loja de artigos de R$ 1,99, com cerca de 10 caixas. Para agilizar o cálculo de quanto cada cliente deve pagar ele desenvolveu um tabela que contém o número de itens que o cliente comprou e ao lado o valor da conta. Desta forma a atendente do caixa precisa apenas contar quantos itens o cliente está levando e olhar na tabela de preços. Você foi contratado para desenvolver o programa que monta esta tabela de preços, que conterá os preços de 1 até 50 produtos, conforme o exemplo abaixo:

|Lojas Quase Dois|Tabela de preços
|---|---|
|1|R$ 1,99|
|2|R$ 3,98|
|...|...|
|50|R$ 99,50|

In [3]:
valor_unitario = 1.99
quantidade = 50

print('Lojas quase Dois - Tabela de Preços')
for i in range(1, quantidade + 1):
    print(f'{i} - R$ {valor_unitario * i:.2f}')

Lojas quase Dois - Tabela de Preços
1 - R$ 1.99
2 - R$ 3.98
3 - R$ 5.97
4 - R$ 7.96
5 - R$ 9.95
6 - R$ 11.94
7 - R$ 13.93
8 - R$ 15.92
9 - R$ 17.91
10 - R$ 19.90
11 - R$ 21.89
12 - R$ 23.88
13 - R$ 25.87
14 - R$ 27.86
15 - R$ 29.85
16 - R$ 31.84
17 - R$ 33.83
18 - R$ 35.82
19 - R$ 37.81
20 - R$ 39.80
21 - R$ 41.79
22 - R$ 43.78
23 - R$ 45.77
24 - R$ 47.76
25 - R$ 49.75
26 - R$ 51.74
27 - R$ 53.73
28 - R$ 55.72
29 - R$ 57.71
30 - R$ 59.70
31 - R$ 61.69
32 - R$ 63.68
33 - R$ 65.67
34 - R$ 67.66
35 - R$ 69.65
36 - R$ 71.64
37 - R$ 73.63
38 - R$ 75.62
39 - R$ 77.61
40 - R$ 79.60
41 - R$ 81.59
42 - R$ 83.58
43 - R$ 85.57
44 - R$ 87.56
45 - R$ 89.55
46 - R$ 91.54
47 - R$ 93.53
48 - R$ 95.52
49 - R$ 97.51
50 - R$ 99.50


**Exercício 30**

O Sr. Manoel Joaquim acaba de adquirir uma panificadora e pretende implantar a metodologia da tabelinha, que já é um sucesso na sua loja de 1,99. Você foi contratado para desenvolver o programa que monta a tabela de preços de pães, de 1 até 50 pães, a partir do preço do pão informado pelo usuário, conforme o exemplo abaixo:

Preço do pão: R$ 0,18

|Panificadora Pão de Ontem|Tabela de preços|
|---|---|
|1|R$ 0,18|
|2|R$ 0,36|
|...|...|
|50|R$ 9,00|


In [1]:
preco_do_pao = 0.18
quantidade = 50

print('Lojas quase Dois - Tabela de Preços')
for i in range(1, quantidade + 1):
    print(f'{i} - R$ {preco_do_pao * i:.2f}')

Lojas quase Dois - Tabela de Preços
1 - R$ 0.18
2 - R$ 0.36
3 - R$ 0.54
4 - R$ 0.72
5 - R$ 0.90
6 - R$ 1.08
7 - R$ 1.26
8 - R$ 1.44
9 - R$ 1.62
10 - R$ 1.80
11 - R$ 1.98
12 - R$ 2.16
13 - R$ 2.34
14 - R$ 2.52
15 - R$ 2.70
16 - R$ 2.88
17 - R$ 3.06
18 - R$ 3.24
19 - R$ 3.42
20 - R$ 3.60
21 - R$ 3.78
22 - R$ 3.96
23 - R$ 4.14
24 - R$ 4.32
25 - R$ 4.50
26 - R$ 4.68
27 - R$ 4.86
28 - R$ 5.04
29 - R$ 5.22
30 - R$ 5.40
31 - R$ 5.58
32 - R$ 5.76
33 - R$ 5.94
34 - R$ 6.12
35 - R$ 6.30
36 - R$ 6.48
37 - R$ 6.66
38 - R$ 6.84
39 - R$ 7.02
40 - R$ 7.20
41 - R$ 7.38
42 - R$ 7.56
43 - R$ 7.74
44 - R$ 7.92
45 - R$ 8.10
46 - R$ 8.28
47 - R$ 8.46
48 - R$ 8.64
49 - R$ 8.82
50 - R$ 9.00


**Exercícios 31**

O Sr. Manoel Joaquim expandiu seus negócios para além dos negócios de 1,99 e agora possui uma loja de conveniências. Faça um programa que implemente uma caixa registradora rudimentar. O programa deverá receber um número desconhecido de valores referentes aos preços das mercadorias. Um valor zero deve ser informado pelo operador para indicar o final da compra. O programa deve então mostrar o total da compra e perguntar o valor em dinheiro que o cliente forneceu, para então calcular e mostrar o valor do troco. Após esta operação, o programa deverá voltar ao ponto inicial, para registrar a próxima compra. A saída deve ser conforme o exemplo abaixo:

Lojas Tabajara  
Produto 1: R$ 2.20  
Produto 2: R$ 5.80  
Produto 3: R$ 0  
Total: R$ 9.00  
Dinheiro: R$ 20.00  
Troco: R$ 11.00  
...

In [3]:
produtos = 0
soma = 0

while True:
    encerrar = input("Digite 0 para encerrar a operação ou qualquer tecla para continuar: ")
    if encerrar == "0":
        break
    while True:
        produtos += 1
        preco = float(input(f'Informe o preço do produto {produtos} (0 para parar): '))
        if preco == 0:
            break
        soma += preco
        print(f'Produto {produtos}: R$ {preco:.2f}')

    print(f'Total: R$ {soma:.2f}')
    dinheiro = float(input('Valor que o cliente forneceu: '))
    troco = dinheiro - soma
    print(f'Troco: R$ {troco:.2f}')
    produtos = 0

Produto 1: R$ 13.00
Total: R$ 13.00
Troco: R$ 12.00
Produto 1: R$ 26.00
Produto 2: R$ 23.00
Total: R$ 62.00
Troco: R$ 8.00


**Exercícios 32**

Faça um programa que calcule o fatorial de um número inteiro fornecido pelo usuário. Ex.: 5!=5.4.3.2.1=120. A saída deve ser conforme o exemplo abaixo:

Fatorial de: 5  
5! =  5 . 4 . 3 . 2 . 1 = 120


In [17]:
n = 5
fatorial = 1

print(f'{n}! = ', end=' ')
for i in range(n, 0, -1):
    fatorial *= i
    if i > 1:
        print(f'{i} .', end=' ')
    else:
        print(i, end=' ')

print(f'= {fatorial}')


5! =  5 . 4 . 3 . 2 . 1 = 120


**Exercícios 33**

O Departamento Estadual de Meteorologia lhe contratou para desenvolver um programa que leia as um conjunto indeterminado de temperaturas, e informe ao final a menor e a maior temperaturas informadas, bem como a média das temperaturas.

**Exercícios 34**

Os números primos possuem várias aplicações dentro da Computação, por exemplo na Criptografia. Um número primo é aquele que é divisível apenas por um e por ele mesmo. Faça um programa que peça um número inteiro e determine se ele é ou não um número primo.

**Exercícios 35**

Encontrar números primos é uma tarefa difícil. Faça um programa que gera uma lista dos números primos existentes entre 1 e um número inteiro informado pelo usuário.

**Exercício 36**

Desenvolva um programa que faça a tabuada de um número qualquer inteiro que será digitado pelo usuário, mas a tabuada não deve necessariamente iniciar em 1 e terminar em 10, o valor inicial e final devem ser informados também pelo usuário, conforme exemplo abaixo:

Montar a tabuada de: 5  
Começar por: 4  
Terminar em: 7  

Vou montar a tabuada de 5 começando em 4 e terminando em 7:  
5 X 4 = 20  
5 X 5 = 25  
5 X 6 = 30  
5 X 7 = 35  

Obs: Você deve verificar se o usuário não digitou o final menor que o inicial.

**Exercício 37**

Uma academia deseja fazer um senso entre seus clientes para descobrir o mais alto, o mais baixo, a mais gordo e o mais magro, para isto você deve fazer um programa que pergunte a cada um dos clientes da academia seu código, sua altura e seu peso. O final da digitação de dados deve ser dada quando o usuário digitar 0 (zero) no campo código. Ao encerrar o programa também deve ser informados os códigos e valores do clente mais alto, do mais baixo, do mais gordo e do mais magro, além da média das alturas e dos pesos dos clientes

**Exercício 38**

Um funcionário de uma empresa recebe aumento salarial anualmente: Sabe-se que:
- Esse funcionário foi contratado em 1995, com salário inicial de R$ 1.000,00;
- Em 1996 recebeu aumento de 1,5% sobre seu salário inicial;
- A partir de 1997 (inclusive), os aumentos salariais sempre correspondem ao dobro do percentual do ano anterior. 

Faça um programa que determine o salário atual desse funcionário. Após concluir isto, altere o programa permitindo que o usuário digite o salário inicial do funcionário.

**Exercício 39**

Faça um programa que leia dez conjuntos de dois valores, o primeiro representando o número do aluno e o segundo representando a sua altura em centímetros. Encontre o aluno mais alto e o mais baixo. Mostre o número do aluno mais alto e o número do aluno mais baixo, junto com suas alturas.

**Exercício 40**

Foi feita uma estatística em cinco cidades brasileiras para coletar dados sobre acidentes de trânsito. Foram obtidos os seguintes dados:
- Código da cidade;
- Número de veículos de passeio (em 1999);
- Número de acidentes de trânsito com vítimas (em 1999). 

Deseja-se saber:  
a) Qual o maior e menor índice de acidentes de transito e a que cidade pertence;  
b) Qual a média de veículos nas cinco cidades juntas;  
c) Qual a média de acidentes de trânsito nas cidades com menos de 2.000 veículos de passeio.

**Exercícios 41**

Faça um programa que receba o valor de uma dívida e mostre uma tabela com os seguintes dados: valor da dívida, valor dos juros, quantidade de parcelas e valor da parcela.

Os juros e a quantidade de parcelas seguem a tabela abaixo:

|Quantidade de Parcelas|% de Juros sobre o valor inicial da dívida|
|---|---|
|1|0|
|3|10|
|6|15|
|9|20|
|12|25|

Exemplo de Saída do programa:
|Valor da Dívida|Valor dos Juros|Quantidade de Parcelas|Valor da Parcela|
|---|---|---|---|
|R$ 1.000,00|0|1|R$ 1.000,00|
|R$ 1.100,00|100|3|R$ 366,00|
|R$ 1.150,00|150|6|R$ 191,67|

**Exercício 42**

Faça um programa que leia uma quantidade indeterminada de números positivos e conte quantos deles estão nos seguintes intervalos: [0-25], [26-50], [51-75] e [76-100]. A entrada de dados deverá terminar quando for lido um número negativo.

**Exercício 43**

O cardápio de uma lanchonete é o seguinte:

|Especificação|Código|Preço|
|---|---|---|
|Cachorro Quente|100|R$ 1,20|
|Bauru Simples|101|R$ 1,30|
|Bauru com ovo|102|R$ 1,50|
|Hambúrguer|103|R$ 1,20|
|Cheeseburguer|104|R$ 1,30|
|Refrigerante|105|R$ 1,00|

Faça um programa que leia o código dos itens pedidos e as quantidades desejadas. Calcule e mostre o valor a ser pago por item (preço * quantidade) e o total geral do pedido. Considere que o cliente deve informar quando o pedido deve ser encerrado.

**Exercício 44**

Em uma eleição presidencial existem quatro candidatos. Os votos são informados por meio de código. Os códigos utilizados são:

1 , 2, 3, 4  - Votos para os respectivos candidatos  
(você deve montar a tabela ex: 1 - Jose/ 2- João/etc)  
5 - Voto Nulo  
6 - Voto em Branco

Faça um programa que calcule e mostre:
- O total de votos para cada candidato;
- O total de votos nulos;
- O total de votos em branco;
- A percentagem de votos nulos sobre o total de votos;
- A percentagem de votos em branco sobre o total de votos. 

Para finalizar o conjunto de votos tem-se o valor zero.

**Exercício 45**

Desenvolver um programa para verificar a nota do aluno em uma prova com 10 questões, o programa deve perguntar ao aluno a resposta de cada questão e ao final comparar com o gabarito da prova e assim calcular o total de acertos e a nota (atribuir 1 ponto por resposta certa). Após cada aluno utilizar o sistema deve ser feita uma pergunta se outro aluno vai utilizar o sistema. Após todos os alunos terem respondido informar:

a) Maior e Menor Acerto;  
b) Total de Alunos que utilizaram o sistema;  
c) A Média das Notas da Turma.


Gabarito da Prova:

01 - A  
02 - B  
03 - C  
04 - D  
05 - E  
06 - E  
07 - D  
08 - C  
09 - B  
10 - A  

Após concluir isto você poderia incrementar o programa permitindo que o professor digite o gabarito da prova antes dos alunos usarem o programa.

**Exercício 46**

Em uma competição de salto em distância cada atleta tem direito a cinco saltos. No final da série de saltos de cada atleta, o melhor e o pior resultados são eliminados. O seu resultado fica sendo a média dos três valores restantes. Você deve fazer um programa que receba o nome e as cinco distâncias alcançadas pelo atleta em seus saltos e depois informe a média dos saltos conforme a descrição acima informada (retirar o melhor e o pior salto e depois calcular a média). Faça uso de uma lista para armazenar os saltos. Os saltos são informados na ordem da execução, portanto não são ordenados. O programa deve ser encerrado quando não for informado o nome do atleta. A saída do programa deve ser conforme o exemplo abaixo:

Atleta: Rodrigo Curvêllo  

Primeiro Salto: 6.5 m  
Segundo Salto: 6.1 m  
Terceiro Salto: 6.2 m  
Quarto Salto: 5.4 m  
Quinto Salto: 5.3 m  

Melhor salto:  6.5 m  
Pior salto: 5.3 m  
Média dos demais saltos: 5.9 m  

Resultado final:  
Rodrigo Curvêllo: 5.9 m

**Exercício 47**

Em uma competição de ginástica, cada atleta recebe votos de sete jurados. A melhor e a pior nota são eliminadas. A sua nota fica sendo a média dos votos restantes. Você deve fazer um programa que receba o nome do ginasta e as notas dos sete jurados alcançadas pelo atleta em sua apresentação e depois informe a sua média, conforme a descrição acima informada (retirar o melhor e o pior salto e depois calcular a média com as notas restantes). As notas não são informados ordenadas. Um exemplo de saída do programa deve ser conforme o exemplo abaixo:

Atleta: Aparecido Parente  
Nota: 9.9  
Nota: 7.5  
Nota: 9.5  
Nota: 8.5  
Nota: 9.0  
Nota: 8.5  
Nota: 9.7  

Resultado final:  
Atleta: Aparecido Parente  
Melhor nota: 9.9  
Pior nota: 7.5  
Média: 9,04

**Exercício 48**

Faça um programa que peça um numero inteiro positivo e em seguida mostre este numero invertido.

Exemplo:  
12376489 => 98467321

**Exercício 49**

Faça um programa que mostre os n termos da Série a seguir:

$$ S = \frac{1}{2} + \frac{2}{3} + \frac{3}{5} + \frac{4}{7} + \frac{5}{9} + ... + \frac{n}{m} $$

Imprima no final a soma da série.



**Exercício 50**

Faça um programa que calcule o valor de H com N termos.

$$ H = 1 + \frac{1}{2} + \frac{1}{3} + \frac{1}{4} + ... + \frac{1}{N} $$